### PR TITLE
Update the Coordinated Vulnerability Disclosure and the security charter

### PR DESCRIPTION
**Pull request by abstractj**

### CODE REVIEW

**Comment by abstractj** [comment]
@keycloak/maintainers @keycloak/security as soon as we agree about it, I will reopen https://github.com/keycloak/keycloak-web/pull/497.

File: SECURITY.md [hunk 1]
```diff
@@ -1,18 +1,38 @@
...
+This policy supports our open approach and is intended to give security researchers clear guidelines for submitting and coordinating discovered vulnerabilities with us. In complying with this policy, you authorize CNCF to work with you to understand and resolve the issue quickly. For more details about our processes, please read the [security charter](SECURITY_CHARTER.md).
+
+## Guidelines
+
+* Research shared with any Keycloak representatives/individual will be reported to and managed by the Keycloak Security Response Team in order to be officially protected and coordinated.  
+* Access and visibility to research and all CVE related data will follow the principle of least privilege by all vendors involved.  
+* Establish and set a reasonable amount of time to resolve the issue before a vulnerability is disclosed publicly; agree and coordinate on public disclosure dates when possible.   
+* Public disclosure should be prioritized on the need to keep company, government, and individual data confidential and the general public safe.  
+* All vendors will honor disclosure/embargo requests in good faith as long as all guidelines are met  
+* NDA signatures are not required
```
Comment (jonkoops) [inline SECURITY.md]:
```suggestion
* All vendors will honor disclosure/embargo requests in good faith as long as all guidelines are met.
* NDA signatures are not required.
```

File: SECURITY_CHARTER.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+* Identification: the Response Team identifies relevant engineers from affected areas and temporarily includes them in private communication channels (e.g., repositories, email threads), forming a temporary Resolution Team.  
+* Efficiency: to prevent accidental disclosure, the Resolution Team remains as small as necessary.  
+* Autonomy: the Resolution Team has the autonomy to involve additional parties such as release coordinators, QE, and documentation teams. Communication with the Response Team is advised when in doubt.  
+* Access Revocation: post-release, access to sensitive communication channels is revoked to uphold the principle of least privilege.
+
+## Process Overview
+
+1. A new vulnerability is reported to the Keycloak security mailing list.  
+2. A CVE ID is assigned.
```
Comment (jonkoops) [inline SECURITY_CHARTER.md]: This should only happen after triage by the Security Response Team. I guess it's implicit here, by might be good to mention.

**Review by jonkoops — APPROVED** [review]
Looks great Bruno, thanks for formalizing this!

File: SECURITY_CHARTER.md (hunk 2 above)
Comment (abstractj) [reply SECURITY_CHARTER.md]: That's implicit, but no problem about explicitly mentioning it.

**Review by rmartinc — APPROVED** [review]
LGTM!

**Review by jonkoops — APPROVED** [review]

**Comment by stianst** [comment]
I don't think this repo is the right place for these. It's like putting them in a hidden directory no one will know about.

**Comment by abstractj** [comment]
@keycloak/maintainers anyone else would like to review?

File: SECURITY.md [hunk 3]
```diff
@@ -1,18 +1,38 @@
-# Security Policy
```
Comment (stianst) [inline SECURITY.md:1]: Heading should be `Security Policy`, `Coordinated Vulnerability Disclosure` should be a sub-heading, maybe even a sub-heading to `Reporting a suspected vulnerability`.

Perhaps we should also consider moving `Supported Versions` to the top.

File: SECURITY_CHARTER.md
Comment (stianst) [inline SECURITY_CHARTER.md:1]: This won't be visible to anyone. `.github` is a special and kinda hidden repository. It works for `SECURITY.md` as that is inherited by all repositories, but that doesn't work for any file.

File: SECURITY.md (hunk 3 above)
Comment (abstractj) [reply SECURITY.md:1]: @stianst does it sound any better now? If not, would you like to make those changes?

File: SECURITY_CHARTER.md
Comment (abstractj) [reply SECURITY_CHARTER.md:1]: @stianst would make more sense to move it to the main repository (keycloak/keycloak)? Maybe under the docs folder?

**Review by stianst — APPROVED** [review]

File: SECURITY_CHARTER.md
Comment (stianst) [reply SECURITY_CHARTER.md:1]: Not really, let's just leave it here for now.

**Pull request merged**